### ISSUE TITLE
Incorrect `reportCallIssue` error when using argument unpacking

### ISSUE THREAD

**Issue by james-thunes** [microsoft/pyright#9880]
**Describe the bug**
When using argument unpacking when calling a function, pyright incorrectly shows a `reportCallIssue` error.

This happens when unpacking from a list (or a set), but *not* from a tuple.

I would expect that there would be no error seen using argument unpacking with either object type.

**Code or Screenshots**
```python
def foo(x, y, z):
  print(f"{x}, {y}, {z}")

foo(*(1, 2), 3) # <-- no error seen

foo(*[1, 2], 3) # <-- ERROR SEEN: pylance reportCallIssue: expected 3 positional arguments
```

**VS Code extension or command-line**
using vscode. ms-python.vscode-pylance version 2025.2.1

**Comment by erictraut** [comment]
This behavior is intended, so I don't consider this a bug.

In the case of a tuple, the type system tracks the number of elements. The type of expression `(1, 2)` is `tuple[Literal[1], Literal[2]]`. In the case of lists, the type system does not track the number of elements because lists are mutable. The type of the expression `[1, 2]` is `list[int]`. 

When unpacking an iterable of indeterminate length within an argument list, pyright assumes that it supplies values for all remaining positional arguments. I'm not 100% convinced this is the most desirable behavior, but it's a convention that was established by mypy, another type checker in the Python ecosystem. I could make arguments both for and against this convention. 

We have been working for the past year to [standardize behaviors across type checkers](https://htmlpreview.github.io/?https://github.com/python/typing/blob/main/conformance/results/results.html). The behavior of an unpack operator within an argument expression is one of those areas that is not yet covered by the [Python typing spec](https://typing.readthedocs.io/en/latest/spec/index.html). I'd like to see this included in the spec at some point.

Once the typing community comes to a consensus on the desired behavior, I'll update pyright to match the spec. In the meantime, I'm going to leave the current behavior in place for consistency with mypy.

As a workaround, you can switch from a list to a tuple expression.

**Comment by james-thunes** [comment]
ah, that makes sense. I don't necessarily *agree* with the behavior, but I can see why unpacking a mutable object may be a problem. Interestingly, I can still get it complain about argument unpacking if I give the function a tuple that's constructed from a list. That's also not unexpected I suppose.

```python
def fun(x,y,z):
    print(x,y,z)

def test():
    varList = [1,2]
    fun(*varList, 3) # <-- error as explained above
    varTuple = (v for v in varList) # create a tuple from the list
    fun(*varTuple, 3) # <-- also an error
```
